contracts/contracts/token/OUSD.sol

```diff
@@ -511,93 +491,181 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
-            .divPrecisely(_rebasingCreditsPerToken)
-            .add(nonRebasingSupply);
+        totalSupply =
+            ((_rebasingCredits * 1e18) / _rebasingCreditsPerToken) +
+            nonRebasingSupply;
 
         emit TotalSupplyUpdatedHighres(
-            _totalSupply,
+            totalSupply,
             _rebasingCredits,
             _rebasingCreditsPerToken
         );
     }
+
+    function delegateYield(address from, address to) external onlyGovernor {
+        require(from != address(0), "Zero from address not allowed");
+        require(to != address(0), "Zero to address not allowed");
+
+        require(from != to, "Cannot delegate to self");
+        require(
+            yieldFrom[to] == address(0) &&
+                yieldTo[to] == address(0) &&
+                yieldFrom[from] == address(0) &&
+                yieldTo[from] == address(0),
+            "Blocked by existing yield delegation"
+        );
+        RebaseOptions stateFrom = rebaseState[from];
+        RebaseOptions stateTo = rebaseState[to];
+
+        require(
+            stateFrom == RebaseOptions.NotSet ||
+                stateFrom == RebaseOptions.StdNonRebasing ||
+                stateFrom == RebaseOptions.StdRebasing,
+            "Invalid rebaseState from"
+        );
+
+        require(
+            stateTo == RebaseOptions.NotSet ||
+                stateTo == RebaseOptions.StdNonRebasing ||
+                stateTo == RebaseOptions.StdRebasing,
+            "Invalid rebaseState to"
+        );
+
+        if (alternativeCreditsPerToken[from] == 0) {
+            _rebaseOptOut(from);
+        }
+        if (alternativeCreditsPerToken[to] > 0) {
+            _rebaseOptIn(to);
+        }
+
+        uint256 fromBalance = balanceOf(from);
+        uint256 creditsFrom = _balanceToRebasingCredits(fromBalance);
+
+        // Set up the bidirectional links
+        yieldTo[from] = to;
+        yieldFrom[to] = from;
+        rebaseState[from] = RebaseOptions.YieldDelegationSource;
+        rebaseState[to] = RebaseOptions.YieldDelegationTarget;
+
+        // Local
+        _creditBalances[from] = fromBalance;
+        alternativeCreditsPerToken[from] = 1e18;
+        _creditBalances[to] += creditsFrom;
+
+        _adjustGlobals(creditsFrom.toInt256(), -fromBalance.toInt256());
+    }
+
+    function undelegateYield(address from) external onlyGovernor {
+        // Require a delegation, which will also ensure a valid delegation
+        require(yieldTo[from] != address(0), "Zero address not allowed");
+
+        address to = yieldTo[from];
+        uint256 fromBalance = balanceOf(from);
+        // these are credits of from account if it were rebasing
+        uint256 creditsFrom = _balanceToRebasingCredits(fromBalance);
+
+        // Remove the bidirectional links
+        yieldFrom[to] = address(0);
+        yieldTo[from] = address(0);
+        rebaseState[from] = RebaseOptions.StdNonRebasing;
+        rebaseState[to] = RebaseOptions.StdRebasing;
```

Comment:
As of now, our default for contracts is `StdNonRebasing`. We plan to delegate yields to only incentives contracts at least as of now. So, for consistency, I'd prefer if this is set to `StdNonRebasing` here as well. However, it's not a big deal. Or it could actually be a parameter to this function as well

Reply:
IMHO, it is fine to leave the `to` account in a `StdRebasing` state after yield delegation stops since the yield delegation targets will, by design, need to be able to handle the token rebasing. (which doesn't hold true for yield delegation source accounts).
Aslo before we start delegating yield we "force" the source accounts to become rebasing. And understanding`undelegateYield` as opposite of `delegateYield` it feels right to me to leave them rebasing.

Reply:
Sounds good to me